src/library_promise.js

```diff
@@ -169,6 +175,43 @@ mergeInto(LibraryManager.library, {
     });
 #if RUNTIME_DEBUG
     dbg('create: ' + id);
+#endif
+    return id;
+  },
+
+  emscripten_promise_all_settled__deps: ['$promiseMap', '$idsToPromises'],
+  emscripten_promise_all_settled: function(idBuf, resultBuf, size) {
+    var promises = idsToPromises(idBuf, size);
+#if RUNTIME_DEBUG
+    dbg('emscripten_promise_all_settled: ' + promises);
+#endif
+    var id = promiseMap.allocate({
+      promise: Promise.allSettled(promises).then((results) => {
+        if (resultBuf) {
+          for (var i = 0; i < size; i++) {
+            var baseOffset = i * {{{ C_STRUCTS.em_settled_result_t.__size__ }}};
+            var resultOffset =
+                baseOffset + {{{ C_STRUCTS.em_settled_result_t.result }}};
+            var valueOffset =
+                baseOffset + {{{ C_STRUCTS.em_settled_result_t.value }}};
+            if (results[i].status === 'fulfilled') {
+              var fulfill = {{{ cDefs.EM_PROMISE_FULFILL }}};
+              {{{ makeSetValue('resultBuf', 'resultOffset', 'fulfill', 'i32') }}};
+              {{{ makeSetValue('resultBuf', 'valueOffset', 'results[i].value', '*') }}};
+            } else {
+              var reject = {{{ cDefs.EM_PROMISE_REJECT }}};
+              {{{ makeSetValue('resultBuf', 'resultOffset', 'reject', 'i32') }}};
+              // Closure can't type `reason` in some contexts
+              /** @suppress {checkTypes} */
+              {{{ makeSetValue('resultBuf', 'valueOffset', 'results[i].reason', '*') }}};
```

Comment:
So these promises only ever get rejected or fullfilled with Numbers?    What happens if one of them is rejected to fulfilled with something that is not a number?

Reply:
Yeah, they're always fulfilled or rejected with pointers because the fulfill values and reject reasons are always represented as `void*` in the C API.

Reply:
I think you can do something like this: `var reason = /** @type {number} */ results[i].reason;`

Reply:
Is there a good way to generalize that for wasm64?

Reply:
We use `number` for pointer in wasm64 and wasm32 so it should just work.